### ISSUE TITLE
[ascend] tl.load masked-out lanes still issue GM memory transactions — SMMU fault when the address is unmapped

### ISSUE THREAD

**Issue by jianzs** [triton-lang/triton-ascend#2332]
## Summary

On triton-ascend (3.2.2), the address of a lane is still dereferenced even when its `mask` evaluates to `False` in `tl.load`. The mask is applied to the returned **value** (correct `other`), but the GM memory transaction is still issued for masked-out lanes. This violates Triton semantics ("a masked-out lane does not access memory") and causes a **device-side SMMU fault / `vector core exception`** whenever such an address falls on an unmapped page.

In practice, whether a program crashes depends on whether the out-of-range addresses happen to be mapped: in many process-local runs the buffers sit inside mapped-but-unused allocator space so the bug is silent, while in a production vLLM serving process (16 NPUs, eager mode) the fault fires deterministically and kills all 16 workers.

## Minimal reproducer (single NPU, no external dependencies)

```python
import torch
import torch_npu  # noqa: F401
import triton
import triton.language as tl


@triton.jit
def masked_load_below_base(x_ptr, o_ptr, BIG: tl.constexpr, BLOCK: tl.constexpr):
    offs = tl.arange(0, BLOCK)
    idx = offs - BIG          # first BIG lanes point *below* x_ptr and are all masked out
    mask = idx >= 0
    v = tl.load(x_ptr + idx, mask=mask, other=0.0)
    tl.store(o_ptr + offs, v)


x = torch.ones(100000, dtype=torch.float32, device="npu:0")
o = torch.zeros(256, dtype=torch.float32, device="npu:0")
masked_load_below_base[(1,)](x, o, 2, 256)
torch.npu.synchronize()       # => rtDeviceSynchronizeWithTimeout: vector core exception
```

Run with page-precise VA mapping so that the memory just below `x_ptr` is **unmapped**:

```bash
PYTORCH_NPU_ALLOC_CONF=expandable_segments:True python3 repro.py
```

### Observed vs expected

| `BIG` | allocator | result |
|---|---|---|
| 0 (all lanes in-bounds) | expandable_segments | OK (`o` all ones) |
| 2 (2 lanes at `x_ptr - 8B`) | expandable_segments | **SMMU fault / vector core exception** |
| 2 | default caching allocator | OK, and **values are masked correctly** (`o[0] == 0`) |
| 1048576 (all lanes masked, ≈ -4 MB) | expandable_segments | **SMMU fault** |
| 1048576 | default caching allocator | OK, `o` all zeros |

Device log for the fault runs (same signature as our production crash):

```
[ERROR] ... [device_error_core_proc.cc:775] CheckSmmuFault:Get isSmmuFault is 1.
[ERROR] ... [davinci_kernel_task.cc:1210] PrintErrorInfoForDavinciTask:[DFX_INFO]
        AI Core kernel execution failed ... fault kernel_name=masked_load_below_base
The extend info: errcode:(0x800000, 0, 0) errorStr: MTE accesses an invalid GM address
        or the cross-device memory access times out.
EE9999: rtDeviceSynchronizeWithTimeout execution failed, reason=vector core exception
```

Interpretation of the matrix:

- With `big=2` and the default allocator, the output values are correct (`other=0` applied per mask), but the same kernel faults once the mapping is tightened with `expandable_segments` — the only thing that changed is the **mapping state** of the addresses computed for masked-out lanes.
- With `big=1048576`, *every* lane is masked out, and it still faults: the access cannot be explained by any actual data read. The backend must be issuing transactions for lanes whose predicate is false.

## Where this hit us in practice

We found this via a deterministic 16-worker crash of a vLLM (v0.26.0-based, native NPU serving path under development) OpenAI API server running a large hybrid (Mamba/GDN + MoE) model on 16 × Ascend 910 in eager mode. The faulting kernel is the **unmodified upstream** `vllm/model_executor/layers/mamba/ops/causal_conv1d.py::_causal_conv1d_fwd_kernel`. Its "sequence shorter than `state_len`" branch builds exactly the pointer pattern above:

```python
VAL = state_len - seqlen                      # e.g. 3 - 1 = 2
x_ptrs = (
    x_base[None, :]
    + ((idx_tokens_conv - VAL) * stride_x_token)[:, None]   # negative offsets for leading lanes
)  # [BLOCK_M, BLOCK_N]
mask_x = (
    (idx_tokens_conv - VAL >= 0)[:, None]
    & (idx_tokens_conv - VAL < seqlen)[:, None]
    & (idx_feats < dim)[None, :]
)
loaded_x = tl.load(x_ptrs, mask_x, 0.0)     # leading lanes are masked, but their addresses are still issued
```

For the first sequence of a packed batch (`sequence_start_index == 0`, `x_base == x_ptr`) these lanes point below the tensor base; in the serving process that page was unmapped, so every worker hit:

```
fault kernel_name=_causal_conv1d_fwd_kernel, program id=149
CheckSmmuFault:Get isSmmuFault is 1
errcode:(0x800000, 0, 0) MTE accesses an invalid GM address
```

We verified exhaustively that the kernel's indices/metadata are all in range and rule out the caller side (details omitted for brevity — e.g. rebuilding every index tensor inside the kernel with identical contents still crashed; rebuilding the data tensors into fresh allocations did not). Clamping those offsets to `>= 0` (which changes nothing semantically, since the lanes are masked out anyway) makes the deterministic crash disappear (2/2 vs 4/4 crash on the unmodified kernel).

## Expected behavior

For `tl.load`/`tl.store` with a `mask`, lanes whose mask is `False` must not generate any memory transaction — matching upstream Triton semantics and the behavior of other backends.

## Suggested investigation

Check the lowering of `tl.load`/`tl.store` in `triton/backends/ascend/` (the mask/predicate handling on the path down to the NPU MTE instructions). The reproducer above can be used as a regression test: it `torch.npu.synchronize()`s cleanly when `BIG=0` with all allocators, and currently faults with `BIG>0` under `PYTORCH_NPU_ALLOC_CONF=expandable_segments:True`.

## Environment details

 - Hardware: Ascend 910 (A3 series)
 - CANN: 9.1.0
 - OS: Linux (aarch64), Python 3.12.13
 - torch: 2.10.0+cpu
 - torch_npu: 2.10.0.post4
 - triton (imported at runtime): 3.2.0
 - triton-ascend (installed package): 3.2.2, backend files under `site-packages/triton/backends/ascend/`

**Comment by hongziqi** [comment]
Thank you for your feedback. This is an issue with DiscreteMaskAccessConversion Pass, and we will fix it as soon as possible
